index.js

```diff
@@ -1,51 +1,37 @@
-const fetch = require('isomorphic-fetch');
+const http = require('http');
+const request = require('request');
 const _ = require('lodash');
 
-let client;
+let settings;
 
-const createQuery = (query, variables) => {
-  const body = {
-    query,
-    variables,
-    operationName: null,
-  };
-
-  return JSON.stringify(body);
-};
-
-const checkResponseStatus = (response) => {
-  if (response.ok) {
-    return response;
-  }
-
-  const error = new Error(response.statusText);
-  error.response = response;
-  throw error;
-};
-
-const parseResponse = response => response.json();
-
-module.exports.register = (opts, dependencies, next) => { // eslint-disable-line consistent-return
+module.exports.register = (opts, dependencies, next) => {
   if (!opts.serverUrl) {
     return next(new Error('The serverUrl parameter is invalid'));
   }
 
-  client = (options, headers, timeout) => fetch(opts.serverUrl, {
```

Comment:
moving this to the execute() as that gets garbage collected on any component's domain execution is finished